Dockerfile

```diff
@@ -26,12 +35,115 @@ RUN git clone https://github.com/njoy/NJOY2016 /opt/NJOY2016 && \
     mkdir build && cd build && \
     cmake -Dstatic=on .. && make 2>/dev/null && make install
 
-# Clone and install OpenMC
-RUN git clone --recurse-submodules https://github.com/openmc-dev/openmc.git \
-    /opt/openmc && cd /opt/openmc && mkdir -p build && cd build && \
-    cmake -Doptimize=on -DHDF5_PREFER_PARALLEL=on .. && \
-    make && make install && \
-    cd .. && pip install -e .[test]
+# Clone and install OpenMC without DAGMC
+RUN if [ "$include_dagmc" = "false" ] ; \
+    then git clone --recurse-submodules https://github.com/openmc-dev/openmc.git ; \
+    /opt/openmc ; \
+    cd /opt/openmc ; \
+    mkdir -p build ; \
+    cd build ; \
+    cmake -Doptimize=on -DHDF5_PREFER_PARALLEL=on .. ; \
+    make ; \
+    make install ; \
+    cd ..  ; \
+    pip install -e .[test] ; \
+    fi
+
+# install addition packages required for DAGMC
+RUN if [ "$include_dagmc" = "true" ] ; \
+    then apt-get --yes install libeigen3-dev ; \
+    apt-get --yes install libblas-dev ; \
+    apt-get --yes install liblapack-dev ; \
+    apt-get --yes install libnetcdf-dev ; \
+    apt-get --yes install libtbb-dev ; \
+    apt-get --yes install libglfw3-dev ; \
+    fi
+
+# Clone and install Embree
+RUN if [ "$include_dagmc" = "true" ] ; \
+    then git clone https://github.com/embree/embree ; \
+    cd embree ; \
+    mkdir build ; \
+    cd build ; \
+    cmake .. -DCMAKE_INSTALL_PREFIX=.. \
+        -DEMBREE_ISPC_SUPPORT=OFF ; \
+    make ; \
+    make install ; \
+    fi
+
+# Clone and install MOAB
+RUN if [ "$include_dagmc" = "true" ] ; \
+    then pip install --upgrade numpy cython ; \
+    mkdir MOAB ; \
+    cd MOAB ; \
+    mkdir build ; \
+    git clone  --single-branch --branch develop https://bitbucket.org/fathomteam/moab/ ; \
+    cd build ; \
+    cmake ../moab -DENABLE_HDF5=ON \
+                -DENABLE_NETCDF=ON \
+                -DBUILD_SHARED_LIBS=OFF \
+                -DENABLE_FORTRAN=OFF \
+                -DCMAKE_INSTALL_PREFIX=/MOAB ; \
+    make ; \
+    make install ; \
+    rm -rf * ; \
+    cmake ../moab -DBUILD_SHARED_LIBS=ON \
+                -DENABLE_HDF5=ON \
+                -DENABLE_PYMOAB=ON \
+                -DENABLE_BLASLAPACK=OFF \
+                -DENABLE_FORTRAN=OFF \
+                -DCMAKE_INSTALL_PREFIX=/MOAB ; \
+    make ; \
+    make install ; \
+    cd pymoab ; \
+    bash install.sh ; \
+    python setup.py install ; \
```

Comment:
I'm going to look into this more. `pymoab` is working just fine in the Docker image. It's nice to know that this could be a viable alternative for installing that package.

Reply:
Thanks for looking at the options here. I didn't get around to trying -DPYMOAB_PREFIX or PYTHONPATH yesterday but whatever you recommend or the current method sounds fine to me

Reply:
So I looked into this a bit more. As it stands the Dockerfile installs `pymoab` a total of three times in two different locations. Once with the `make` command, another with `install.sh`, and yet again with the `python setup.py install` command.

There's no harm in it other than being potentially confusing or winding up importing a `pymoab` version you weren't expecting to if a reinstall of MOAB is performed in the image. Overall, it would be nice to simplify that so it only exists in one place. To do that there are two options I can suggest:

  1. Set the`PYMOAB_INSTALL_PREFIX` CMake variable directly in the CMake step.

    This allows you to specify the exact location to install the module. `/usr/local/lib/python3.8/dist-packages` in this case.

  2. Install MOAB to the default location without a `CMAKE_INSTALL_PREFIX`. This will default to the system python modules directory for the `pymoab` installation.

Either of those solutions should install `pymoab` to the system location with the `make install` command.

Reply:
I can see pymoab is installing when the two ```-DCMAKE_INSTALL_PREFIX=\MOAB \``` lines are removed from the MOAB install. 

I'm wondering what to put in the next stage of the dockerfile which builds double-down, this needs to know the ```-DMOAB_DIR```

I tried setting ```-DMOAB_DIR=/MOAB/build/``` but this appears to have not worked in other ways.

![Screenshot from 2020-12-25 19-48-57](https://user-images.githubusercontent.com/8583900/103141303-830eaa00-46ea-11eb-900f-8222611c12c2.png)

Reply:
If an installation prefix is not specified, CMake will install to `/usr/local/` by default. So `-DMOAB_DIR=/usr/local` should do the trick.

Reply:
Thanks for that, appears to work just fine over here